### PR TITLE
Use v1beta1 bundle-release.yaml with v1.0.1 provider tags and clusterctl to create v1alpha3 clusters

### PR DESCRIPTION
**Pull request by mrajashree**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/810
https://github.com/aws/eks-anywhere/issues/815

*Description of changes:*
This is just so we can merge the upgrade to v1.0.1 tags/v1beta1 api version in stages.
This PR adds the following:
* A feature toggle such that if env var `V1BETA1_BUNDLE` is set to `true`, `make eks-a` uses the bundle-release.yaml from v1beta1-experimental branch
* Adds following changes for using capi v1.0.1 tag:
   * Remove usage of `watching-namespace` flag from clusterctl since its removed upstream
   * Webhooks for all providers run within the same pod now, so remove wait for webhook pods
* Adds following changes for using capv v1.0.1 tag:
   * CAPV v1.0.1 expects cloud-controller-manager to be managed through ClusterResourceSet. This PR updates the capv manifest, adding cloud-controller-manager manifest and adding the required components to ClusterResourceSet
   * Removes patching of cloud-controller-manager to add tolerations for taints on controlplane nodes, since these tolerations are now directly applied on the cloud-controller-manager spec when it is deployed
 
We should be able to continue using v1alpha3 manifests with clusterctl v1.0.1 until other changes [from this](https://github.com/aws/eks-anywhere/pull/750/commits) mainly code changes to start using v1beta1 and v1beta1 CAPV/CAPD manifests are not merged.
I have tested creating a stacked and an external etcd ubuntu cluster (not BR) of v1alpha3 version using v1.0.1 clusterctl tag through the changes in this PR.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/clustermanager/internal/deployments.go [hunk 1]
```diff
@@ -16,23 +13,6 @@ var ExternalEtcdDeployments = map[string][]string{
 	"etcdadm-bootstrap-provider-system": {"etcdadm-bootstrap-provider-controller-manager"},
 }
 
-// map between file name and the capi/v deployments
-var ClusterDeployments = map[string]*types.Deployment{
```
Comment (mrajashree) [inline pkg/clustermanager/internal/deployments.go:20]: I was going to remove the webhook pods from this, but it shows up as an unused var so thought I'll remove it

File: pkg/providers/vsphere/config/template-cp.yaml [hunk 2]
```diff
@@ -964,3 +972,252 @@ kind: ConfigMap
...
+          serviceAccountName: cloud-controller-manager
+          tolerations:
+          - effect: NoSchedule
+            key: node.cloudprovider.kubernetes.io/uninitialized
+            value: "true"
+          - effect: NoSchedule
+            key: node-role.kubernetes.io/master
+          - effect: NoSchedule
+            key: node.kubernetes.io/not-ready
+{{- if .controlPlaneTaints }}
```
Comment (mrajashree) [inline pkg/providers/vsphere/config/template-cp.yaml:1203]: this is the part that does what we currently do fetching cloud-controller-manager and patching it with tolerations here: https://github.com/aws/eks-anywhere/blob/main/pkg/providers/vsphere/vsphere.go#L1232-L1262

File: pkg/providers/vsphere/vsphere.go [hunk 3]
```diff
@@ -1226,37 +1238,38 @@ func (p *vsphereProvider) RunPostControlPlaneUpgrade(ctx context.Context, oldClu
...
-			return p.providerKubectlClient.ApplyTolerationsFromTaintsToDaemonSet(ctx, oldClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints, clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints, cloudControllerDaemonSetContainerName, workloadCluster.KubeconfigFile)
-		},
-	)
-	if err != nil {
-		return fmt.Errorf("failed to apply tolerations on VSphere cloud controller manager daemonset post upgrade: %v", err)
+		// Step 2: Patch DaemonSet vsphere-cloud-controller-manager in namespace kube-system with toleration values for the taints provided in
+		// the cluster spec file
+		err = p.Retrier.Retry(
+			func() error {
+				return p.providerKubectlClient.ApplyTolerationsFromTaintsToDaemonSet(ctx, oldClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints, clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints, cloudControllerDaemonSetContainerName, workloadCluster.KubeconfigFile)
```
Comment (bnrjee) [inline pkg/providers/vsphere/vsphere.go:1267]: Do we need this ? I can see that the older taints are being hard coded.

Comment (vivek-koppuru) [reply pkg/providers/vsphere/vsphere.go:1267]: This is just preserving the older logic

**Comment by bnrjee** [comment]
/lgtm
Make sure you have tested the upgrade scenario and vsphere cloud controller manager deployment has appropriate toleration added.

**Comment by vivek-koppuru** [comment]
/unhold

tested create and upgrade with taints and they work now

**Comment by vivek-koppuru** [comment]
/hold

testing changes after rebase

**Pull request merged**